raiden_libs/network/matrix/client.py

```diff
@@ -21,31 +23,59 @@ class GMatrixHttpApi(MatrixHttpApi):
     A wrapper around MatrixHttpApi to limit the number
     of concurrent requests we make to the number of connections
     available to us in requests.Session connection pool size.
+
+    Args:
+        pool_maxsize: max size of underlying/session connection pool
+        retry_timeout: for how long should a single request be retried if it errors
+        retry_delay: delay to way between a retries inside a request. If a callable,
+            receives the previous delay and iteration number
+            and should return the new delay
     """
     def __init__(
             self,
             *args,
-            max_retries: int = 3,
             pool_maxsize: int = 10,
+            retry_timeout: int = 60,
+            retry_delay: Union[float, Callable[[float, int], float]] = 1,
```

Comment:
Why not just a generator constructor? `Callable[None, Iterator[float]]` is easier to understand IMO

Reply:
I wanted a stateless delay calculation function. But yes, I suppose a generator would work as good, I'll change it to keep it compatible with the other codes we use generators to generate increasing/exponential delays.